urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/com/kokakiwi/mclauncher/ui/simple/components/TexturedPanel.java

```diff
@@ -0,0 +1,74 @@
+package com.kokakiwi.mclauncher.ui.simple.components;
+
+import java.awt.Color;
+import java.awt.GradientPaint;
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.Image;
+import java.awt.geom.Point2D;
+
+import javax.imageio.ImageIO;
+import javax.swing.JPanel;
+
+public class TexturedPanel extends JPanel
+{
+    private static final long serialVersionUID = -1577652681975914598L;
+    private Image             img;
+    private final Image       bgImage;
+    
+    public TexturedPanel(String path) throws Exception
+    {
+        
+        this(ImageIO.read(TexturedPanel.class.getResourceAsStream("/" + path))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *getResourceAsStream*. Currently, there are execution paths that do not contain closure statements, for example, when ImageIO.read throws an exception. Ensure the object returned by *getResourceAsStream()* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).